exercises/08.permissions/04.solution.utils/app/utils/permissions.ts

```diff
@@ -40,7 +40,14 @@ export async function requireUserWithPermission(
 	return user.id
 }
 
-export async function requireUserWithRole(request: Request, name: string) {
+const roles = {
+	user: 'user',
+	admin: 'admin'
+} as const
+
+type Role = keyof typeof roles
```

Comment:
I think we could simplify this a lot by doing this instead:
```suggestion
type Role = 'user' | 'admin'
```